clients/iceberg-extensions/src/test/java/org/projectnessie/spark/extensions/ITNessieStatements.java

```diff
@@ -0,0 +1,365 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.spark.extensions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.projectnessie.client.tests.AbstractSparkTest;
+import org.projectnessie.error.BaseNessieClientServerException;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.ContentsKey;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.model.ImmutableCommitMeta;
+import org.projectnessie.model.ImmutableOperations;
+import org.projectnessie.model.Operation;
+import org.projectnessie.model.Operations;
+import org.projectnessie.model.Tag;
+
+public class ITNessieStatements extends AbstractSparkTest {
+
+  private String hash;
+  private final String refName = "testBranch";
+
+  @BeforeAll
+  protected static void createDelta() {
+    conf.set(
+        "spark.sql.extensions", "org.projectnessie.spark.extensions.NessieSparkSessionExtensions");
+  }
+
+  @BeforeEach
+  void getHash() {
+    try {
+      hash = nessieClient.getTreeApi().getDefaultBranch().getHash();
+    } catch (NessieNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @AfterEach
+  void removeBranches() throws NessieConflictException, NessieNotFoundException {
+    for (String s : new String[] {refName, "main"}) {
+      try {
+        nessieClient
+            .getTreeApi()
+            .deleteBranch(s, nessieClient.getTreeApi().getReferenceByName(s).getHash());
+      } catch (BaseNessieClientServerException e) {
+        // pass
+      }
+    }
+    nessieClient.getTreeApi().createReference(Branch.of("main", null));
+  }
+
+  @Test
+  void testCreateBranchIn() throws NessieNotFoundException {
+
+    List<Object[]> result = sql("CREATE BRANCH %s IN nessie", refName);
+    assertEquals("created branch", Collections.singletonList(row("Branch", refName, hash)), result);
+    Assertions.assertEquals(
+        nessieClient.getTreeApi().getReferenceByName(refName), Branch.of(refName, hash));
+    result = sql("DROP BRANCH %s IN nessie", refName);
+    assertEquals("deleted branch", Collections.singletonList(row("OK")), result);
```

Comment:
nit: should we create an overloaded method for `AbstractSparkTest#assertEquals()` that accepts a single `Row` and puts it internally into a `List`? Then you could just do `assertEquals("deleted branch", row("OK"), result)` instead of having to put everything into a `Collections.singletonList` yourself

Reply:
fixed